crates/typst-library/src/text/shaping.rs

```diff
@@ -488,17 +488,24 @@ impl<'a> ShapedText<'a> {
         }
 
         // Find any glyph with the text index.
-        let mut idx = self
-            .glyphs
-            .binary_search_by(|g| {
-                let ordering = g.range.start.cmp(&text_index);
-                if ltr {
-                    ordering
-                } else {
-                    ordering.reverse()
-                }
-            })
-            .ok()?;
+        let cmp = |g: &ShapedGlyph| {
+            let ordering = g.range.start.cmp(&text_index);
+            if ltr {
+                ordering
+            } else {
+                ordering.reverse()
+            }
+        };
```

Comment:
I think it would be more readable to extract the full `self.glyphs.binary_search_by(|..| ..)` into a variable than just the comparison function.